oap-ws/src/main/java/oap/ws/WsService.java

```diff
@@ -135,89 +137,119 @@ public void handle( Request request, Response response ) {
-                                            case HEADER:
-                                                return convert( parameter.name(), parameter.type(),
-                                                    request.header( parameter.name() ) );
-                                            case PATH:
-                                                return wsMethod.map( wsm -> convert( parameter.name(), parameter.type(),
-                                                    ServiceUtil.pathParam( wsm.path(), request.requestLine,
-                                                        parameter.name() ) ) )
-                                                    .orElseThrow( () -> new WsException(
-                                                        "path parameter " + parameter.name() + " without " +
-                                                            WsMethod.class.getName() + " annotation" ) );
-                                            case BODY:
-                                                return parameter.type().assignableFrom( byte[].class ) ?
-                                                    ( parameter.type().isOptional() ? request.readBody() :
-                                                        request.readBody()
-                                                            .orElseThrow( () -> new WsClientException(
-                                                                "no body for " + parameter.name() ) )
-                                                    ) :
-                                                    convert( parameter.name(), parameter.type(), request.readBody() );
-                                            default:
-                                                return parameter.type().assignableTo( List.class ) ?
-                                                    convert( parameter.type(),
-                                                        request.parameters( parameter.name() ) ) :
-                                                    convert( parameter.name(), parameter.type(),
-                                                        request.parameter( parameter.name() ) );
-
-                                        }
-                                    } )
-                                    .orElseGet( () -> parameter.type().assignableTo( List.class ) ?
-                                        convert( parameter.type(), request.parameters( parameter.name() ) ) :
-                                        convert( parameter.name(), parameter.type(),
-                                            request.parameter( parameter.name() ) )
-                                    );
-
-                                paramErrors.addAll( validators.forParameter( parameter, impl )
-                                    .validate( value ) );
-
-                                paramValues[i] = value;
-                            }
+                        if( !sessionAware ) {
```

Comment:
This has to be refactored in future (most probably the whole process of handling)